pkg/provisioner/ironic/ironic.go

```diff
@@ -652,6 +653,8 @@ func (p *ironicProvisioner) UpdateHardwareState() (result provisioner.Result, er
 		discoveredVal = true
 	case powerOff:
 		discoveredVal = false
+	case softPowerOff:
+		discoveredVal = false
```

Comment:
this is too simplistic. There are more places where you need to handle a soft power off instead of a power off. like line 1334. you should also account for the delay after which you need to trigger an actual power off.